### PR TITLE
Dspace bat file

### PR DESCRIPTION
**Pull request by Flusspferd123**
Removes the "The entered line is too long. Syntax error" error when entering dspace script commands in Windows 10.  E.g. "dspace create-administrator"

### CODE REVIEW

**Comment by AlexanderS** [comment]
Should we remove `bin/buildpath.bat`? It may be a left over from ancient java versions when the 8.3 naming convention was required.

Note: You removed `dspace/config/dspace.cfg` and `dspace/config/modules/authentication.cfg`. Maybe you can rebase your changes, so that you only include the commit "Minor changes to bat file"?

**Comment by Flusspferd123** [comment]
I also think that bin/buildpath.bat can be removed. Shall i remove it from this PR?

**Comment by tdonohue** [comment]
@Flusspferd123 : Yes, since this PR seems to be removing the (prior) purpose of `buildpath.bat`, then I'd recommend you remove it in this PR.  Otherwise, it's completely unused, but still sitting in the codebase.

**Comment by abollini** [comment]
it could be useful to rework also the "linux" dspace script in a similar way as it will be useful to have a shorter command line also here. Moreover, it is better to keep the two scripts as close as possible

BTW: I just noted that the dspace script still mention an OAI webapp classpath that of course not longer exist

**Comment by Flusspferd123** [comment]
Please give this an extra carefull review. The "linux" dspace script now works on a Linux machine. The bat file works for windows. If gitBash is used on Windows the linux" dspace script does not work.
Please also notice: An empty space in the path of a dependency will break this.

File: dspace/bin/dspace [hunk 1]
```diff
@@ -12,20 +12,18 @@
 
 BINDIR=`dirname $0`
 DSPACEDIR=`cd "$BINDIR/.." ; pwd`
-# Add the directory with all oai classes (original and overlay) needed by the 'dspace oai' launcher.xml
-OAICLASSES=$DSPACEDIR/webapps/oai/WEB-INF/classes/
 
 # Get the JARs in $DSPACEDIR/jsp/WEB-INF/lib, separated by ':'
-JARS=`echo $DSPACEDIR/lib/*.jar | sed 's/ /\:/g'`
+JARS="$DSPACEDIR/lib/*"
```
Comment (mwoodiupui) [inline dspace/bin/dspace]: This gives a space-separated list, not the colon-separated path-list needed by -classpath.
The ``sed`` command could be replaced with ``tr ' ' ':'``

Comment (Flusspferd123) [reply dspace/bin/dspace]: That is correct, but what about spaces in folder names? They would be replaced as well, causing Jars to not appear on the classpath.

Comment (mwoodiupui) [reply dspace/bin/dspace]: True:  we can't use simple string replacement.  But we must have a colon-separated list.
The path will have to be built iteratively, like this:
```
unset JARS
for JAR in ${DSPACEDIR}/lib/*
do
    [ -v JARS ] && JARS=${JARS}:
    JARS=${JARS}${JAR}
done
```
and then put doublequotes around $FULLPATH to protect any interior spaces.

Comment (mwoodiupui) [reply dspace/bin/dspace]: I did some testing and found (as I should have known) that `JARS="$DSPACEDIR/lib/*" is not expanded, so we would be passing the glob expression into `java` as the classpath.  At least in OpenJDK, the interpreter seems to be expanding the glob itself, contrary to its own documentation.  I recommend that we not depend on undocumented behavior.

**Review by abollini — APPROVED** [review]
Tested on linux with opendjk. It works,

File: dspace/bin/dspace.bat [hunk 2]
```diff
@@ -26,7 +26,7 @@ echo Using DSpace installation in: %cd%
 
 REM Build a CLASSPATH including all classes in oai webapp, all libraries in [dspace]/lib and the config folder.
 set DSPACE_CLASSPATH=%CLASSPATH%;config;webapps\oai\WEB-INF\classes\
```
[the remark this answers is not in the document]
Comment (tdonohue) [reply dspace/bin/dspace.bat]: Agreed, the OAI webapp libraries are no longer needed here (they don't exist anymore).  This (and the line before) can be changed to:
```
REM Build a CLASSPATH including all libraries in [dspace]/lib and the config folder.
set DSPACE_CLASSPATH=%CLASSPATH%;config
```

**Review by tdonohue — APPROVED** [review]
👍 I've tested this on Windows 10 and it works well.  However, there's one minor thing left to fix.  As noted below, the OAI webapp classes can be removed from the `DSPACE_CLASSPATH`.  Once that is fixed, this can be merged.

@Flusspferd123: can you make this one last change?  Once that's completed, we can merge this.

**Comment by tdonohue** [comment]
I've made the tiny one-line change that was requested in this PR above (see [these comments](https://github.com/DSpace/DSpace/pull/2544#discussion_r347086718)). This PR is already at +2 and was only sitting open waiting on that minor cleanup.

So, this can be merged as soon as Travis approves again.

File: dspace/bin/dspace.bat (hunk 2 above)
[the remark this answers is not in the document]
Comment (tdonohue) [reply dspace/bin/dspace.bat]: I've made this update and pushed to this PR.

**Pull request merged**